packages/flutter/lib/src/cupertino/picker.dart

```diff
@@ -102,8 +126,8 @@ class CupertinoPicker extends StatefulWidget {
   /// listen for [ScrollEndNotification] and read its [FixedExtentMetrics].
   final ValueChanged<int> onSelectedItemChanged;
 
-  /// [Widget]s in the picker's scroll wheel.
-  final List<Widget> children;
+  /// A delegate that helps lazily instantiating child.
+  final SliverChildDelegate childDelegate;
```

Comment:
Ummm, I'm a bit conflicted here (nothing of your fault). Sliver is a specific thing which we're not using here so it's confusing that we're referring to slivers directly here. There's nothing too sliver specific in SliverChildDelegate and its subclasses yet (beyond what could just be a LazyListChildDelegate). But there's also a risk that we'd add sliver specific things into SliverChildDelegate in the future and make it incompatible here. 

Thoughts @Hixie? 

I'm thinking we can extract a LazyChild(List/Builder)Delegate out as supers of SliverChild(List/Builder)Delegate and use those here (since you're not using estimateMaxScrollOffset and didFinishLayout anyway).

Reply:
If it is about to change in the future, I can write new independent classes following the same pattern for us to use here. Furthermore, because we are planning to have options of looping, negative index, and maybe more, is it better to write new ones?

Reply:
i definitely agree that we shouldn't use something called Sliver.

We want to avoid code duplication where it makes sense. If you can extract a common base class cleanly, that's probably worth doing. If it would make it a mess, then having separate classes here makes sense.